flink-dist/src/main/flink-bin/bin/taskmanager.sh

```diff
@@ -63,7 +57,7 @@ if [[ $STARTSTOP == "start" ]]; then
         #
         TM_MAX_OFFHEAP_SIZE="8388607T"
 
-        if [[ "${STREAMINGMODE}" == "batch" ]] && useOffHeapMemory; then
+        if [[ "${FLINK_TM_MEM_ALLOC}" == "eager" ]] && useOffHeapMemory; then
```

Comment:
You might want to allocate lazily in batch jobs with off-heap memory. In the current state of the pull request, we could run into out of memory errors then. On the other hand, I see that we don't want to subtract heap memory in streaming with lazy memory allocation (because we don't allocate managed memory for streaming jobs for now).